### PR TITLE
Add shell job block

### PR DESCRIPTION
**Pull request by ahuang11**
I’ve been migrating over some process utils https://github.com/PrefectHQ/prefect/blob/main/src/prefect/utilities/processutils.py and not directly importing it because if I remember correctly, the function does too much all at once, but if we are to follow the JobBlock paradigm of trigger, wait, and fetch, it needs to be separated.

also, we need to make sure powershell is also supported https://github.com/PrefectHQ/prefect-shell/blob/main/prefect_shell/commands.py#L67-L73

This https://github.com/PrefectHQ/prefect-shell/pull/54/files should be handled automatically by copying over the process utils

Lastly, I think there was a problem where I stream the output (which prints to screen), but upon completion, the output returned None so I think we need to append to list

Closes #

### Checklist

- [ ] This pull request references any related issue by including "Closes #<ISSUE_NUMBER>"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-shell/issues/new/choose) first.
- [ ] This pull request includes tests or only affects documentation.
 - [ ] Summarized PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-shell/blob/main/CHANGELOG.md)

### CODE REVIEW

**Comment by ahuang11** [comment]
Found a way to use Prefect's built-in implementation which supports Windows.

File: prefect_shell/commands.py [hunk 1]
```diff
@@ -110,3 +121,95 @@ def example_shell_run_command_flow():
...
+
+    @sync_compatible
+    async def fetch_result(self) -> List[str]:
+        """
+        Retrieve the results of the job run and return them.
+        """
+        return self._output
+
+
+class ShellJob(JobBlock):
```
Comment (desertaxle) [inline prefect_shell/commands.py:163]: I think we need a different name for this. Maybe something like `ShellOperation`?

Also, I think this block could feasibly have a method that handles all three steps in one call, but there's probably multiple ways we could expose that functionality.

**Review by desertaxle — COMMENTED** [review]
This is looking good so far! Left a couple of comments on things that stood out to me on first review.

**Comment by ahuang11** [comment]
image

image

File: prefect_shell/commands.py [hunk 2]
```diff
@@ -110,3 +118,284 @@ def example_shell_run_command_flow():
...
+
+        Examples:
+            Sleep for 5 seconds and then print "Hello, world!":
+            ```python
+            from prefect_shell import ShellOperation
+
+            with ShellOperation(
+                commands=["sleep 5", "echo 'Hello, world!'"],
+            ) as shell_operation:
+                shell_process = await shell_operation.trigger()
```
Comment (desertaxle) [inline prefect_shell/commands.py:284]: For consistency
```suggestion
                shell_process = shell_operation.trigger()
```

File: prefect_shell/commands.py [hunk 3]
```diff
@@ -110,3 +118,284 @@ def example_shell_run_command_flow():
...
+            )
+        )
+        self.logger.info(
+            f"PID {process.pid} triggered with {num_commands} commands running "
+            f"inside the {(self.working_dir or '.')!r} directory."
+        )
+        return ShellProcess(shell_operation=self, process=process)
+
+    @sync_compatible
+    async def run(self, **open_kwargs: Dict[str, Any]) -> List[str]:
```
Comment (desertaxle) [inline prefect_shell/commands.py:338]: For this method, can we do the context management ourselves? It seems like a bit much to have a context manager for a short-running command or script.

File: prefect_shell/commands.py [hunk 4]
```diff
@@ -110,3 +118,284 @@ def example_shell_run_command_flow():
...
+        ```python
+        from prefect_shell import ShellOperation
+
+        shell_operation = ShellOperation.load("BLOCK_NAME")
+        ```
+    """
+
+    _block_type_name = "Shell Operation"
+    _logo_url = "https://raw.githubusercontent.com/PrefectHQ/prefect-shell/main/docs/images/logo.png"  # noqa: E501
+    _documentation_url = "https://images.ctfassets.net/gm98wzqotmnx/3Nn6D7zVqrdIsGjufgpES4/abd12fa6427121f7363758aafc468d9a/Terminalicon2.png?h=250"  # noqa: E501
```
Comment (desertaxle) [inline prefect_shell/commands.py:224]: This URL points to an image. Should this be for `_logo_url`?

**Review by desertaxle — CHANGES_REQUESTED** [review]

**Review by desertaxle — APPROVED** [review]

**Pull request merged**